control-plane/catalog/registration/cache.go

```diff
@@ -148,38 +228,51 @@ func (c *RegistrationCache) updateTermGWACLRole(log logr.Logger, registration *v
 		return err
 	}
 
-	roles, _, err := client.ACL().RoleList(nil)
-	if err != nil {
-		log.Error(err, "error reading role list")
-		return err
+	var data bytes.Buffer
+	if err := gatewayTpl.Execute(&data, templateArgs{
+		EnablePartitions: c.partitionsEnabled,
+		Partition:        registration.Spec.Service.Partition,
+		EnableNamespaces: c.namespacesEnabled,
+		Namespace:        registration.Spec.Service.Namespace,
+		ServiceName:      registration.Spec.Service.Name,
+	}); err != nil {
+		// just panic if we can't compile the simple template
+		// as it means something else is going severly wrong.
+		panic(err)
 	}
 
-	policy := &capi.ACLPolicy{
-		Name:        servicePolicyName(registration.Spec.Service.Name),
-		Description: "Write policy for terminating gateways for external service",
-		Rules:       fmt.Sprintf(`service %q { policy = "write" }`, registration.Spec.Service.Name),
-		Datacenters: []string{registration.Spec.Datacenter},
-		Namespace:   registration.Spec.Service.Namespace,
-		Partition:   registration.Spec.Service.Partition,
-	}
+	var mErr error
+	for _, termGW := range termGWsToUpdate {
+		// the terminating gateway role is _always_ in the default namespace
+		roles, _, err := client.ACL().RoleList(&capi.QueryOptions{})
+		if err != nil {
+			log.Error(err, "error reading role list")
+			return err
+		}
 
-	existingPolicy, _, err := client.ACL().PolicyReadByName(policy.Name, nil)
-	if err != nil {
-		log.Error(err, "error reading policy")
-		return err
-	}
+		policy := &capi.ACLPolicy{
+			Name:        servicePolicyName(registration.Spec.Service.Name),
+			Description: "Write policy for terminating gateways for external service",
+			Rules:       data.String(),
+			Datacenters: []string{registration.Spec.Datacenter},
+		}
 
-	if existingPolicy == nil {
-		policy, _, err = client.ACL().PolicyCreate(policy, nil)
+		existingPolicy, _, err := client.ACL().PolicyReadByName(policy.Name, &capi.QueryOptions{})
 		if err != nil {
-			return fmt.Errorf("error creating policy: %w", err)
+			log.Error(err, "error reading policy")
+			return err
 		}
-	} else {
-		policy = existingPolicy
-	}
 
-	var mErr error
-	for _, termGW := range termGWsToUpdate {
+		writeOpts := &capi.WriteOptions{}
```

Comment:
Do policies and roles have global or partition scope (as opposed to a particular namespace)?

Reply:
Discussed on a call that it would be helpful to comment why these don't include a namespace. This is because historically all roles + policies are just created in the `default` namespace, and we don't want to re-evaluate that decision in the scope of this work.